theme: Update light theme to match Epiphany from GNOME 45

garrett opened the pull request:
GNOME 45 is releasing soon and has a few style changes, which also affect Epiphany too.

https://wiki.gnome.org/FortyFive

- Beta: 5 Aug
- RC: 2 Sep (tomorrow, as of writing this)
- Release: 20 Sep

I've started adjusting colors where I've seen changes, but I'm sure I'm missing a few. So far, it's starting to look pretty close for both the standard light mode and private browsing.

Dark mode doesn't look like it has changed, but I haven't looked too closely.

garrett commented:
Here's a screenshot:

![image](https://github.com/rafaelmardojai/firefox-gnome-theme/assets/10246/9182de5f-bb34-47ba-9138-e4aceeb22151)

One thing I've noticed that I haven't acommodated yet: There's a lack of a divider bar between the toolbar and tab bar. I'm not sure if it should also be removed between that and the bookmarks bar too.

There's still a bar between the top UI elements and the page:

![image](https://github.com/rafaelmardojai/firefox-gnome-theme/assets/10246/f03e079f-cdc9-4f24-9f7c-c2806baee0bc)

rafaelmardojai commented on theme/colors/light.css:
```diff
@@ -103,9 +103,9 @@
-	--gnome-button-background: color-mix(in srgb, currentColor 10%, transparent);
-	--gnome-button-hover-background: color-mix(in srgb, currentColor 15%, transparent);
-	--gnome-button-active-background: color-mix(in srgb, currentColor 30%, transparent);
+	--gnome-button-background: #ebebeb;
+	--gnome-button-hover-background: #e0e0e0;
+	--gnome-button-active-background: #dedede;
```
We don't want opaque colors here, otherwise when we recolor the headebar in private browsing it will look bad.

Currently colors are base on the libadwaita 1.3 ones. If colors have changed we must check the libadwaita 1.4 sass files and update the opacity o whatever has changed.

rafaelmardojai replied:
Actually, this is not used by flat buttons (they use `--gnome-hover-color`), but the point is the same, we want to use translucent colors when libadwaita uses them.

garrett replied:
Thanks! Good to know. I'll try to see if I can find time to update it further this weekend.

garrett replied:
I reverted those specific color changes. After the other color changes, those wind up being the correct colors anyway. (I checked with a screenshot afterward and compared with Epiphany.)

garrett commented:
After fixing up the hardcoded color issue, I removed the separators between parts of the UI at have the bar only where the UI meets the page, exactly how Epiphany does it in GNOME 45.

garrett commented:
I've been using this on my work and personal computers for the past couple weeks and it seems fine. It's probably ready (or near ready) for merging, unless someone notices an issue. I'm happy to adjust anything if someone points out any issues.

GNOME 45 releases in one week from today, so merging it between now and then probably makes sense.

garrett commented:
That all said; I just noticed one small problem: There is extra space as the result of the dividers. But now, as the dividers have been removed, the extra space should be too.

This PR:

![image](https://github.com/rafaelmardojai/firefox-gnome-theme/assets/10246/7013f7a2-1b8a-46c4-866b-284326b2ad87)

Upcoming Epiphany:

![image](https://github.com/rafaelmardojai/firefox-gnome-theme/assets/10246/114c1923-258f-4422-93c7-a4d4a1197653)

It's a subtle few pixels of vertical space. I suppose the padding on the top of the bars (except the very top one) should be decreased 4px.

astrocaat commented:
It looks good to me. The only problem I found is that colors in private window are messed up -- it's just a minor thing (at least to me). Look here:
 
![Screenshot from 2023-09-20 08-13-22](https://github.com/rafaelmardojai/firefox-gnome-theme/assets/143425234/4e8953f8-b636-4050-8323-a133940a1d6f)

garrett commented:
@astrocaat: I wonder if that's this specific line: https://github.com/rafaelmardojai/firefox-gnome-theme/pull/662/files#diff-b411e2aa6d7b2cf034533c8e237350f0507b8547765c6712eefbf7d571212020R161 — it's the only new color definition.

rafaelmardojai commented on theme/colors/light.css:
```diff
@@ -144,24 +144,27 @@
-	--gnome-tabbar-tab-active-hover-background: #c0cbd7; /* Hardcoded color */
+	--gnome-tabbar-tab-active-hover-background: #d1d8e1; /* Hardcoded color */
+	/* Buttons */
+	--gnome-button-background: #d7dfe8; /* Hardcoded color */
```
Buttons color should be transparent, default ones should be good for private window. This same color is used for entries backgrounds. Any reason for this change?

CoolingTool commented:
Tried this on Firefox Nightly and for some reason the border between the header bar and the page is transparent
![image](https://github.com/rafaelmardojai/firefox-gnome-theme/assets/62089272/fb206cac-9099-4513-aa12-7be9b5c368ac)

garrett commented:
I've resolved everything above (transparent line in newer firefox, private button color override, and the extra spacing that's no longer needed for the tab bar as there's no separator line in GNOME Web 45). I've also squashed everything down and rebased to the current master branch.

This should be ready for another review. Thanks!

rafaelmardojai commented:
Tabs separators are also misaligned:
![Captura desde 2023-10-26 09-25-53](https://github.com/rafaelmardojai/firefox-gnome-theme/assets/6210397/c73a46cf-8db4-474d-a38e-4aabfcd17104)

When removing the padding from tabs we need to also adjust other elements, tabsbar buttons margin, scroll overlays, etc.

rafaelmardojai commented:
I have a local fix, will push it.

rafaelmardojai commented:
Merged it to a new `gnome45` branch. I will merge it to master when everything is updated (dark theme, etc).

JaneSmith commented:
Is this supposed to be released now? As far as I can tell, the GNOME 45 support was released with v120 on Nov 21? I just updated Firefox (121.0 Flatpak) and my theme, and now I have the issue CodingTool mentioned, where there's a transparent bar below the tab bar. https://github.com/rafaelmardojai/firefox-gnome-theme/pull/662#issuecomment-1741950167
